app/controllers/Application.java

```diff
@@ -1689,6 +1698,168 @@ public static Result restFlowMetricsGraphData(String flowDefId) {
+      if (userSuggestedParamSet == null) {
+        logger.info("No user suggested param set for jobDefinitionId:" + jobDefinitionId);
+      } else {
+        logger.info("User suggested param set: " + userSuggestedParamSet.id);
+      }
+
+      logger.info("Tuning Job Definition Id: " + tuningJobDefinition.job.id);
+      Boolean autoApply = tuningJobDefinition.autoApply;
+
+      TuningAlgorithm tuningAlgorithm = jobSuggestedParamSet.tuningAlgorithm;
+      List<TuningParameter> parametersList = getTuningParametersListForJob(tuningAlgorithm.id);
+      logger.info("Tuning Algorithm id: " + tuningAlgorithm.id);
+      DecimalFormat truncateParamValueFormat = new DecimalFormat("#.##");
+      JsonArray tuningParameters = new JsonArray();
+      for (TuningParameter tuningParam : parametersList) {
+        String paramName = tuningParam.paramName;
+        Integer id = tuningParam.id;
+        JobSuggestedParamValue userSuggestedParameterValue = null;
+        if (userSuggestedParamSet != null) {
+          userSuggestedParameterValue = JobSuggestedParamValue.find.select("*")
+              .where()
+              .eq(JobSuggestedParamValue.TABLE.jobSuggestedParamSet + "." + JobSuggestedParamSet.TABLE.id, userSuggestedParamSet.id)
+              .eq(JobSuggestedParamValue.TABLE.tuningParameter + "." + TuningParameter.TABLE.id, id)
+              .findUnique();
+        }
+
+        JobSuggestedParamValue jobSuggestedParameterValue = JobSuggestedParamValue.find.select("*")
+            .where()
+            .eq(JobSuggestedParamValue.TABLE.jobSuggestedParamSet + "." + JobSuggestedParamSet.TABLE.id, jobSuggestedParamSet.id)
+            .eq(JobSuggestedParamValue.TABLE.tuningParameter + "." + TuningParameter.TABLE.id, id)
+            .findUnique();
+
+        if (jobSuggestedParameterValue == null) {
+          logger.info("No Job Suggested Param Value");
+          continue;
+        }
+        JsonObject param = new JsonObject();
+        param.addProperty("name", paramName);
+        param.addProperty("jobSuggestedParamValue",  truncateParamValueFormat.format(jobSuggestedParameterValue.paramValue));
+        if (userSuggestedParameterValue != null) {
+          param.addProperty("userSuggestedParamValue",  truncateParamValueFormat.format(userSuggestedParameterValue.paramValue));
+          param.addProperty("currentParamValue",  truncateParamValueFormat.format(userSuggestedParameterValue.paramValue));
+        } else {
+          param.addProperty("userSuggestedParamValue",  truncateParamValueFormat.format(jobSuggestedParameterValue.paramValue));
+          param.addProperty("currentParamValue",  truncateParamValueFormat.format(jobSuggestedParameterValue.paramValue));
+        }
+        tuningParameters.add(param);
+        logger.info("param: " + tuningParameters);
+      }
+
+      String currentTuningAlgorithm = tuningAlgorithm.
+          optimizationAlgo.name().contains("PSO") ? "OBT" : "HBT";
```

Comment:
Can you move the constants to some constants file. 
Also, for future changes if this line can be refactored into a method which returns the algorithm name. As of now, this code violates OCP.